### PR TITLE
Make initial modular cli changes

### PR DESCRIPTION
**Pull request by maxjeffos**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [x] Reviewed by Snyk internal team

#### What does this PR do?
Transforms this into a monorepo using NPM Workspaces (a feature of NPM@7) and Lerna (primary for versioning and publishing NPM packages) and, in the process, creates an initial experimental-only `@snyk/protect` package.

#### How could this be manually tested?
Like normal except make sure you use npm@7.

#### Any background context you want to provide?
These are structural changes which will allow us to do a bunch of things, most notably and most imminently, extracting a standalone `@snyk/protect` package which will be used in Snyk Fix PRs instead of the full `snyk` package, providing a much lighter-weight thing for customers to need to use for the patching ("protect") functionality with only the bare minimum of additional dependencies. This package is included in this PR however it is not production-ready code, as indicated at `packages/snyk-protect/README.md`.

The goal is that this new monorepo structure will also allow teams contributing to the CLI to be able to do so more autonomously and have more clear code-ownership.

https://www.notion.so/snyk/Extract-Snyk-Protect-Modular-CLI-a2d9a6a536d34806be69ab85a8ba8b31

#### What are the relevant tickets?
HAMMER-237

### CODE REVIEW

File: packages/cli-protect/LICENSE [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Copyright 2015 Snyk Ltd.
```
Comment (JackuB) [inline packages/cli-protect/LICENSE:1]: Afaik Lerna will actually copy the license from root folder to all subprojects on publish.

Comment (maxjeffos) [reply packages/cli-protect/LICENSE]: Cool, I removed it.

File: .circleci/config.yml [hunk 2]
```diff
@@ -226,13 +237,75 @@ jobs:
       - setup_remote_docker:
           version: 19.03.13
           # docker_layer_caching: true
+      
+      - install_github_cli
+      - run:
+          name: Check npm version
+          command: |
+            npm --version
+            sudo npm install -g npm@7
```
Comment (maxjeffos) [inline .circleci/config.yml:246]: Doing this twice - remove

File: .circleci/config.yml [hunk 3]
```diff
@@ -226,13 +237,75 @@ jobs:
...
+
+      - store_artifacts:
+          path: ./binary-releases
+
+      - run:
+          name: Create a GitHub release
+          command: |
+            latest_version=$(cat lerna.json | jq .version -r)
+            new_tag="v${latest_version}"
+            gh release create ${new_tag} --title "${new_tag}" -n "Release notes not implemented yet" --prerelease
```
Comment (maxjeffos) [inline .circleci/config.yml:291]: --prelease so the first merge to master will only produce a pre-release in GitHub

File: package.json [hunk 4]
```diff
@@ -22,12 +23,16 @@
   "engines": {
     "node": ">=8"
   },
+  "workspaces": [
+    ".",
+    "packages/*"
+  ],
   "scripts": {
-    "build": "tsc",
+    "build": "tsc --build tsconfig.json",
```
Comment (maxjeffos) [inline package.json:31]: make this just `tsc --build`

File: release-scripts/docker-desktop-release.sh [hunk 5]
```diff
@@ -7,6 +7,8 @@ mkdir ./dist/docker/cli
 cp -r ./dist/cli ./dist/docker/cli/cli
 cp -r ./dist/lib ./dist/docker/cli/lib
 cp ./package.json ./dist/docker/
+# cp ./tsconfig.json ./dist/docker/
```
Comment (maxjeffos) [inline release-scripts/docker-desktop-release.sh:10]: delete leftover comments

File: .circleci/config.yml (hunk 2 above)
Comment (maxjeffos) [reply .circleci/config.yml:246]: done

File: package.json (hunk 4 above)
Comment (maxjeffos) [reply package.json:31]: done

File: release-scripts/docker-desktop-release.sh (hunk 5 above)
Comment (maxjeffos) [reply release-scripts/docker-desktop-release.sh:10]: done

**Review by JackuB — APPROVED** [review]

**Pull request merged**